numpy/core/arrayprint.pyi

```diff
@@ -0,0 +1,148 @@
+import sys
+from types import TracebackType
+from typing import Any, Optional, Callable, Union, Type
+
+# Using a private class is by no means ideal, but it is simply a consquence
+# of a `contextlib.context` returning an instance of aformentioned class
+from contextlib import _GeneratorContextManager
+
+from numpy import (
+    ndarray,
+    generic,
+    bool_,
+    integer,
+    timedelta64,
+    datetime64,
+    floating,
+    complexfloating,
+    void,
+    str_,
+    bytes_,
+    longdouble,
+    clongdouble,
+)
+from numpy.typing import ArrayLike, _CharLike, _FloatLike
+
+if sys.version_info > (3, 8):
+    from typing import Literal, TypedDict
+else:
+    from typing_extensions import Literal, TypedDict
+
+class _FormatDict(TypedDict, total=False):
+    bool: Callable[[bool_], str]
+    int: Callable[[integer[Any]], str]
+    timedelta: Callable[[timedelta64], str]
+    datetime: Callable[[datetime64], str]
+    float: Callable[[floating[Any]], str]
+    longfloat: Callable[[longdouble], str]
+    complexfloat: Callable[[complexfloating[Any, Any]], str]
+    longcomplexfloat: Callable[[clongdouble], str]
+    void: Callable[[void], str]
+    numpystr: Callable[[_CharLike], str]
+    object: Callable[[object], str]
+    str: Callable[[Any], str]  # Unused but still present?
```

Comment:
The docstring of `np.setprintoptions` ([doc](https://github.com/numpy/numpy/blob/bac54ecb0b815eea2e8116d21aa96e1387793468/numpy/core/arrayprint.py#L132-L157)) mentions an optional `str` key, though for the past 8 years it has actually been non-functional (see https://github.com/numpy/numpy/pull/459).

Can we just remove any mention of it, both here and in the docstring?

Reply:
Sounds fine to me.

Reply:
Done as of https://github.com/numpy/numpy/pull/18032/commits/82e5747dd40cbbde9c728452f21fa4716492c81d.

Just to stress: the above-mentioned push does not deprecate anything, 
it merely removes a few lines of unused code (and updates the docstring).